quinn-proto/src/connection/mod.rs

```diff
@@ -2351,13 +2360,24 @@ where
                             "RETIRE_CONNECTION_ID for unissued sequence number",
                         ));
                     }
+                    self.cids_active_seq.remove(&sequence);
+                    let allow_more_cid = self
+                        .peer_params
+                        .active_connection_id_limit
+                        .0
+                        .min(LOC_CID_COUNT)
+                        > self.cids_active_seq.len() as u64;
                     self.endpoint_events
-                        .push_back(EndpointEventInner::RetireConnectionId(sequence));
+                        .push_back(EndpointEventInner::RetireConnectionId(
+                            sequence,
+                            allow_more_cid,
+                        ));
```

Comment:
Consider a scenario where peer A has active remote cid 0,1,2. Peer B first send a NEW_CONNECTION_ID with cid 3 and retire_prior_to set to 0. Peer A processes this NEW_CONNECTION_ID frame; update remote cid to 1,2,3 and meanwhile send a RETIRE_CONNECTION_ID from to peer B. If peer B doesn't check the cid limit here and send a new cid again, peer A will then face CONNECTION_ID_LIMIT_ERROR

Reply:
Probably good to have this comment in the code!

Reply:
Done